### DISCUSSION TITLE
batchsize accuracy

### DISCUSSION THREAD

**Discussion by tifffiaany** [BindsNET/bindsnet#688] [Q&A]
When I tried to increase batchsize the accuracy of my network was significantly lower than batchsize=1. What should I do to improve the accuracy?

**Comment by Hananel-Hazan** [comment]
Usually changing batch size require tuning some other parameters (STDP strength, threshold flexibility etc). I suggest running some exploratory search to find the right parameter.

**Reply by tifffiaany** [reply]
Hi, thank you for your reply.
But , I can't find which parameters represent STDP strength and threshold flexibility.  Can you tell me more information about parameters tuning. 
And the learning rule and network used are PostPre and DiehlAndCook2015. Are they suitable for batch_size>1?
Thank you!

**Reply by Hananel-Hazan** [reply]
Thank you for your response.

For tuning parameters like STDP strength and threshold flexibility, you can look into the following:

- **STDP Strength**: This is often represented by parameters like `A_plus` and `A_minus` which control the magnitude of weight changes for potentiation and depression. In BindsNET  `nu`, is a 2-tuple specifying pre- and post-synaptic learning rates.
- **Threshold Flexibility**: This might be controlled by parameters like `v_thresh` which determines the neuron's firing threshold.

Each layer, neuron, and learning rule has parameters that you can change and adapt to find the right configuration. I suggest reading the documentation for the [STDP function](https://bindsnet-docs.readthedocs.io/guide/guide_part_ii.html) for more detailed information.

Regarding the learning rule and network, both PostPre and DiehlAndCook2015 can work with batch sizes greater than 1, but they may require careful tuning. It’s crucial to experiment with different parameter settings to find the optimal configuration for larger batch sizes. You can refer to this [example in BindsNET](https://github.com/BindsNET/bindsnet/blob/master/examples/mnist/batch_eth_mnist.py) for guidance on how to use batch processing.

**Reply by rafaelblevin821** [reply]
You have `exc` and `inh` strength parameters which act as multipliers on the learning rate for excitatory and inhibitory updates. The learning rate (`nu`) is a separate parameter.

Both can be used in tandem to adjust the effectiveness of the learning.

**Reply by tifffiaany** [reply]
Thank you for your reply. I will try to adjust these parameters.
